group_project_v2/mixins.py

```diff
@@ -232,10 +246,58 @@ def studio_view(self, context):  # pylint: disable=unused-argument, no-self-use
         return fragment
 
 
-class DashboardMixin(object):
+class DashboardXBlockMixin(object):
+    """ Mixin for an XBlock that has dashboard views """
     def dashboard_view(self, context):
         raise NotImplementedError(MUST_BE_OVERRIDDEN)
 
+    def dashboard_detail_view(self, context):
+        raise NotImplementedError(MUST_BE_OVERRIDDEN)
+
+
+class DashboardRootXBlockMixin(ProjectAPIXBlockMixin):
+    """
+    Mixin for an XBlock that can act as a root XBlock for dashboard view.
+    Dashboard root XBlock is responsible for injecting workgroups and students into the view context
+    """
+    TARGET_STUDENTS = 'target_students'
+    TARGET_WORKGROUPS = 'target_workgroups'
+
+    def _append_context_parameters_if_not_present(self, context):
+        if self.TARGET_STUDENTS not in context:
+            context[self.TARGET_STUDENTS] = list(self.all_users_in_workgroups)
+
+        if self.TARGET_WORKGROUPS not in context:
+            context[self.TARGET_WORKGROUPS] = list(self.workgroups)
+
+    @property
+    def project_details(self):
+        """
+        Gets ProjectDetails for current block
+        :rtype: group_project_v2.project_api.dtos.ProjectDetails
+        """
+        raise NotImplementedError(MUST_BE_OVERRIDDEN)
+
+    @property
+    def workgroups(self):
+        """
+        :rtype: list[group_project_v2.project_api.dtos.WorkgroupDetails]
+        """
+        workgroup_ids = self.project_details.workgroups
+
+        workgroups = []
+        for workgroup_id in workgroup_ids:
+            workgroups.append(self.project_api.get_workgroup_by_id(workgroup_id))
+
+        return workgroups
```

Comment:
Nit: You could replace these four lines with one line that `return`s a list comprehension.